File: VB/XtraReport1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections
Imports System.Data
Imports System.Drawing.Printing
Imports DevExpress.XtraReports.UI
' ...

Namespace WindowsApplication1
	Partial Public Class XtraReport1
		Inherits XtraReport
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub xrLabel4_PreviewDoubleClick(ByVal sender As Object, ByVal e As PreviewMouseEventArgs) Handles xrLabel6.PreviewDoubleClick, xrLabel5.PreviewDoubleClick, xrLabel4.PreviewDoubleClick
			Dim frm As New Form2()
			Dim label As XRLabel = TryCast(e.Brick.BrickOwner, XRLabel)
			Dim fieldName As String = label.DataBindings("Text").DataMember.Substring(label.DataBindings("Text").DataMember.LastIndexOf("."c) + 1)
			frm.str = (CType(e.Brick.Value, DataRowView))(fieldName).ToString()
			frm.ShowDialog()
			CType(e.Brick.Value, DataRowView)(fieldName) = frm.str
			Me.CreateDocument(False)
		End Sub

		Private Sub xrLabel4_BeforePrint(ByVal sender As Object, ByVal e As PrintEventArgs) Handles xrLabel6.BeforePrint, xrLabel5.BeforePrint, xrLabel4.BeforePrint
			CType(sender, XRLabel).Tag = Me.DetailReport.GetCurrentRow()
		End Sub
	End Class
End Namespace

File: CS/XtraReport1.cs
using System;
using System.Collections;
using System.Data;
using System.Drawing.Printing;
using DevExpress.XtraReports.UI;
// ...

namespace WindowsApplication1 {
    public partial class XtraReport1 : XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }

        private void xrLabel4_PreviewDoubleClick(object sender, PreviewMouseEventArgs e) {
            Form2 frm = new Form2();
            XRLabel label = e.Brick.BrickOwner as XRLabel;
            string fieldName = label.DataBindings["Text"].DataMember.Substring(label.DataBindings["Text"].DataMember.LastIndexOf('.') + 1);
            frm.str = ((DataRowView)e.Brick.Value)[fieldName].ToString();
            frm.ShowDialog();
            ((DataRowView)e.Brick.Value)[fieldName] = frm.str;
            this.CreateDocument(false);
        }

        private void xrLabel4_BeforePrint(object sender, PrintEventArgs e) {
            ((XRLabel)sender).Tag = this.DetailReport.GetCurrentRow();
        }
    }
}
